flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/PostTransformOperator.java

```diff
@@ -362,16 +363,17 @@ private Tuple2<BinaryRecordData, Boolean> transformRecord(
             @Nullable TransformFilterProcessor filterProcessor,
             TransformContext context) {
         RecordData.FieldGetter[] preFieldGetters = info.getPreTransformedFieldGetters();
-        Schema preSchema = info.getPreTransformedSchema();
-        Schema postSchema = info.getPostTransformedSchema();
+
+        List<DataType> preDataTypes = info.getPreTransformedDataTypes();
+        List<DataType> postDataTypes = info.getPostTransformedDataTypes();
+
         BinaryRecordDataGenerator postGenerator = info.getPostTransformedRecordDataGenerator();
 
         Object[] preRow = new Object[preFieldGetters.length];
         for (int i = 0; i < preFieldGetters.length; i++) {
             preRow[i] =
                     DataTypeConverter.convertToOriginal(
-                            preFieldGetters[i].getFieldOrNull(recordData),
-                            preSchema.getColumnDataTypes().get(i));
```

Comment:
I think reducing the time consumption of this `Schema.getColumnDataTypes()` method would be more appropriate, as the caller may not be clear about the logic behind it, and this is an immutable class.